src/nifreeze/model/pet.py

```diff
@@ -130,9 +121,5 @@ def predict(self, index=None, **kwargs):
         # self._coeff is V (num. voxels) x K - 4
         predicted = np.squeeze(A @ self._coeff.T)
 
-        if self._mask is None:
-            return predicted.reshape(self._datashape)
-
-        retval = np.zeros(self._datashape, dtype="float32")
-        retval[self._mask] = predicted
-        return retval
```

Comment:
The mask was not properly applied because the fit is done in the full dataset, it doesn't make sense to only apply the mask at the output. I'll leave fitting only within the mask for a future PR.